docs/configuration.md

```diff
@@ -22,14 +22,24 @@ An example of one that works with the example implementation can be found in [`o
 
 ## Environment variables
 
-In order for the implementation to know where your configuration JSON file is located, you can set an environment variable `OPTIMADE_CONFIG_FILE` with the value of the _full_ path to the JSON file.
+In order for the implementation to know where your configuration JSON file is located, you can set an environment variable `OPTIMADE_CONFIG_FILE` with the either the value of the _absolute_ path to the configuration file or the _relative_ path to the file from the current working directory of where the server is run.
 
 This variable is actually an extension of the configuration option `config_file`.
 By default, the server will try to load a JSON file called `.optimade.json` located in your home folder (or equivalent).
 
 Here the generally recognized environment variable prefix becomes evident, namely `OPTIMADE_` or `optimade_`.
 Hence, you can set (or overwrite) any configuration option from the server's defaults or a value read from the configuration JSON by setting an environment variable named `OPTIMADE_<configuration_option>`.
 
+## Custom configuration options
+
+One can extend the current list of configuration options by sub-classing [`ServerConfig`][optimade.server.config.ServerConfig] and adding configuration options as attributes with values of `Field` (`pydantic.field`).
+Any attribute type will be validated through `pydantic` as is the case for all of the regular configuration options.
+
+This is useful for, e.g., custom database backends, if one wants to utilize the general server configuration setup implemented in `optimade` to declare specific database information.
+It can also be useful if one wishes to extend and build upon the general `optimade` server with new endpoints and routes.
+
+Remember to instantiate an instance of the sub-class, which can be imported and used in your application.
```

Comment:
Do we also have to do something like

```python
from optimade.server.config import CONFIG

class MyConfig(ServerConfig):
    pass

CONFIG = MyConfig(**overrides)
```

to get the submodules that import CONFIG to work correctly?

Reply:
pinging @CasperWA as I don't know if you get notified as it is my PR

Reply:
Not sure if that's an issue. It's not one I've run into with the gateway? Since `optimade.server.config.CONFIG` will only have a subset of the sub-classes config class, it shouldn't be a problem, unless you manually start updating the `CONFIG` attributes in your server's lifetime?

Reply:
I guess this might be a nice thing to do to be "safe", but I'm not sure whether it truly works in all or at least most cases - maybe it does?

Reply:
Yeah I don't do it this way either. In that case lets leave this as is